app/screens/apps_form/apps_form_container.tsx

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+
+import {Theme} from '@mm-redux/types/preferences';
+import {AppCall, AppCallResponse, AppField, AppForm, AppFormValue, AppFormValues, AppLookupCallValues, AppSelectOption} from '@mm-redux/types/apps';
+import {AppCallResponseTypes, AppCallTypes} from '@mm-redux/constants/apps';
+import AppsForm from './apps_form';
+
+export type Props = {
+    form?: AppForm;
+    call?: AppCall;
+    actions: {
+        doAppCall: (call: AppCall) => Promise<{data: AppCallResponse}>;
+    };
+    theme: Theme;
+    componentId: string;
+};
+
+export type State = {
+    form?: AppForm;
+}
+
+const makeError = (errMessage: string) => {
+    return {
+        data: {
+            type: 'error',
+            error: 'There has been an error submitting the modal. Contact the app developer. Details: ' + errMessage,
+        },
+    };
+};
+
+export default class AppsFormContainer extends PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        this.state = {form: props.form};
+    }
+
+    handleSubmit = async (submission: {values: AppFormValues}): Promise<{data: AppCallResponse<any>}> => {
+        //TODO use FormResponseData instead of Any
+        const {form} = this.state;
+        if (!form) {
+            return makeError('submitForm state.form is not defined');
+        }
+
+        const call = this.getCall();
+        if (!call) {
+            return makeError('submitForm props.call is not defined');
```

Comment:
why `props` here and below?

Reply:
getCall gets the call from the props. If it does not exist, it will return null. Therefore there is no call if it is not in props.call. Not really clear I admit.